projectmanagement-domain/src/main/java/life/qbic/projectmanagement/application/policy/directive/InformUsersAboutBatchRegistration.java

```diff
@@ -0,0 +1,118 @@
+import java.util.List;
+import life.qbic.domain.concepts.DomainEvent;
+import life.qbic.domain.concepts.DomainEventSubscriber;
+import life.qbic.domain.concepts.communication.CommunicationService;
+import life.qbic.domain.concepts.communication.Content;
+import life.qbic.domain.concepts.communication.Recipient;
+import life.qbic.domain.concepts.communication.Subject;
+import life.qbic.logging.api.Logger;
+import life.qbic.projectmanagement.application.Messages;
+import life.qbic.projectmanagement.application.authorization.acl.ProjectAccessService;
+import life.qbic.projectmanagement.domain.project.Project;
+import life.qbic.projectmanagement.domain.project.ProjectId;
+import life.qbic.projectmanagement.domain.project.sample.Sample;
+import life.qbic.projectmanagement.domain.project.sample.event.SampleBatchRegistered;
+import life.qbic.user.api.UserInformationService;
+import org.jobrunr.jobs.annotations.Job;
+import org.jobrunr.scheduling.JobScheduler;
+import org.springframework.stereotype.Component;
+
+/**
+ * <b>Directive: Send email to project users about new sample batch</b>
+ * <p>
+ * After a sample batch has been registered, we need to inform users about
+ * the newly registered samples.
+ *
+ * @since 1.0.0
+ */
+@Component
+public class InformUsersAboutBatchRegistration implements DomainEventSubscriber<SampleBatchRegistered> {
+
+  private static final Logger log = logger(InformUsersAboutBatchRegistration.class);
+  private final CommunicationService communicationService;
+  private final ProjectAccessService projectAccessService;
+  private final UserInformationService userInformationService;
+  private final JobScheduler jobScheduler;
+
+  public InformUsersAboutBatchRegistration(CommunicationService communicationService,
+      ProjectAccessService projectAccessService, UserInformationService userInformationService, JobScheduler jobScheduler) {
+    this.projectAccessService = projectAccessService;
+    this.userInformationService = userInformationService;
+    this.communicationService = communicationService;
+    this.jobScheduler = jobScheduler;
+  }
+
+  @Override
+  public Class<? extends DomainEvent> subscribedToEventType() {
+    return SampleBatchRegistered.class;
+  }
+
+  @Override
+  public void handleEvent(SampleBatchRegistered event) {
+    jobScheduler.enqueue(() -> notifyUsersAboutSamples(event.project(), event.samples()));
+  }
+
+  @Job(name = "Notify users about batch registration")
+  /**
+   * Sends an email with attached spreadsheet of newly registered samples to the users of a project.
+   * The email contains an explanation about sample identifiers and the spreadsheet the known metadata of the samples
+   * @param project - the project the sample batch was added to
+   * @param sampleBatch - the samples that were registered
+   */
+  public void notifyUsersAboutSamples(Project project, Collection<Sample> sampleBatch) {
+    List<RecipientDTO> recipientsWithAccess = getUsersWithAccess(project.getId());
```

Comment:
ah sorry, my mistake. The method promises to return users, not recipients. So you might need a static wrapper method like ``userToRecipient(UserInfo user)``